_posts/2017-12-30-Secure-your-architecture-part2.md

```diff
@@ -0,0 +1,223 @@
+    ],
+    "n": "yeNlzlub94YgerT030codqEztjfU_S6X4DbDA_iVKkjAWtYfPHDzz_sPCT1Axz6isZdf3lHpq_gYX4Sz-cbe4rjmigxUxr-FgKHQy3HeCdK6hNq9ASQvMK9LBOpXDNn7mei6RZWom4wo3CMvvsY1w8tjtfLb-yQwJPltHxShZq5-ihC9irpLI9xEBTgG12q5lGIFPhTl_7inA1PFK97LuSLnTJzW0bj096v_TMDg7pOWm_zHtF53qbVsI0e3v5nmdKXdFf9BjIARRfVrbxVxiZHjU6zL6jY5QJdh1QCmENoejj_ytspMmGW7yMRxzUqgxcAqOBpVm0b-_mW3HoBdjQ",
+    "e": "AQAB",
+    "kid": "NjVBRjY5MDlCMUIwNzU4RTA2QzZFMDQ4QzQ2MDAyQjVDNjk1RTM2Qg",
+    "x5t": "NjVBRjY5MDlCMUIwNzU4RTA2QzZFMDQ4QzQ2MDAyQjVDNjk1RTM2Qg"
+  }
+]}
+{% endhighlight %}
+
+Explanation properties:
+
+* `alg`: is the algorithm for the key
+* `kty`: is the key type
+* `use`: is how the key was meant to be used. For the example above sig represents signature.
+* `x5c`: is the x.509 certificate chain
+* `e`: is the exponent for a standard pem
+* `n`: is the modulus for a standard pem
+* `kid`: is the unique identifier for the key
+* `x5t`: is the thumbprint of the x.509 cert (SHA-1 thumbprint)
+
+
+
+## Cloud-hosted Key management service
+KMS is a fully managed service that allows you to manage your encryption keys in the cloud.
+Most of these KMSs offer the best way for encryption and generate, rotate and destroy your keys. 
+But the KMS is vendor lock-in so all your keys will stay on the platform.
+To avoid vendor lock-in, we can implement our own open source version for managing our encryption keys. 
+
+A few examples to get an idea of KMS:
+* [Google Cloud KMS](https://cloud.google.com/kms/){:target="_blank"}
+* [AWS KMS](https://aws.amazon.com/kms/){:target="_blank"}
+* [Azure Vault](https://azure.microsoft.com/en-us/services/key-vault/){:target="_blank"}
+
+A few examples of open-source variants:
+* [Spring Cloud Config Server](https://cloud.spring.io/spring-cloud-config/){:target="_blank"}
+* [Vault](https://www.vaultproject.io/){:target="_blank"}
+* [Keywhiz](https://square.github.io/keywhiz/){:target="_blank"}
+
+
+## Spring Cloud Config Server
+The Spring Cloud Config Server provides a centralized external configuration management backed optionally by a Git repository or database.
+Using a REST API for external configuration, Config Server supports encryption and decryption of properties and yml files. 
+First step is downloading the Java Cryptography Extension on our local pc.
+>JCE provides a framework and implementation for encryption, key generation, key agreement and message authentication code algorithms. 
+You're not installing JCE itself, because it's packaged within the Java SE binary.
+However, you do need to update its policy files from time to time.
+[Downloads](http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html) are available for Java 6, 7 and 8.
```

Comment:
There is a download link for each java version:
6: http://www.oracle.com/technetwork/java/javase/downloads/jce-6-download-429243.html
7: http://www.oracle.com/technetwork/java/javase/downloads/jce-7-download-432124.html
8: http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html